include/vara/Feature/Result.h

```diff
@@ -0,0 +1,115 @@
+#ifndef VARA_FEATURE_RESULT_H
+#define VARA_FEATURE_RESULT_H
+
+#include "llvm/Support/raw_ostream.h"
+
+#include "vara/Utils/VariantUtil.h"
+
+#include <cassert>
+#include <iostream>
+#include <type_traits>
+#include <variant>
+
+namespace vara::feature {
+
+namespace result {
+
+template <typename ValueTy>
+class Ok {
+public:
+  Ok(ValueTy V) : V(std::move(V)) {}
+
+  ValueTy operator*() { return std::move(V); }
```

Comment:
I'm not sure if we should always move out the value, take a look at my proposal below, but also feel free to argu against it.

```suggestion
  ValueTy operator*() && { return extract_value(); }
  ValueTy operator*() & { return V; }
  
  ValueTy extract_value() { return std::move(V); }
```
In cases where we are an `rvalue` we can savely always move out the value, as it will not be needed afterwards.
If we are not, we "by default" copy the value out but offer and extra function `extract_value` to allow the caller to move out the value explicitly.